### PR TITLE
Add support for ed25519 keys (#440)

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/git.rs [hunk 1]
```diff
@@ -902,8 +903,16 @@ impl<'a> RemoteCallbacks<'a> {
...
+                    if ssh_key_paths_to_try.is_none() {
+                        if let Some(ref mut cb) = self.get_ssh_keys {
+                            let mut paths = cb(username);
+                            paths.reverse();
+                            ssh_key_paths_to_try = Some(paths);
+                        }
+                    }
+
+                    if let Some(paths) = &mut ssh_key_paths_to_try {
+                        if let Some(path) = paths.pop() {
```
Comment (lukegb) [inline lib/src/git.rs:915]: Does this mean we try only the first SSH key that exists?

Comment (martinvonz) [reply lib/src/git.rs:915]: It's supposed to mean that we try one after another. Do you think it won't work or were you just curious?

Comment (lukegb) [reply lib/src/git.rs:915]: Oh - whoops, I completely misread this. For some reason I thought ssh_key_paths_to_try was being set up _inside_ the callback. I should really not be reading code at 2am.

File: lib/src/git.rs [hunk 2]
```diff
@@ -902,8 +903,16 @@ impl<'a> RemoteCallbacks<'a> {
                         });
                     }
 
-                    if let Some(ref mut cb) = self.get_ssh_key {
-                        if let Some(path) = cb(username) {
+                    if ssh_key_paths_to_try.is_none() {
+                        if let Some(ref mut cb) = self.get_ssh_keys {
```
Comment (yuja) [inline lib/src/git.rs:907]: nit: `.get_or_insert_with()` could be used. If `get_ssh_keys()` isn't provided, the paths can be initialized to `vec![]`.

**Review by yuja — APPROVED** [review]

File: cli/src/commands/git.rs [hunk 3]
```diff
@@ -508,8 +508,8 @@ fn with_remote_callbacks<T>(ui: &mut Ui, f: impl FnOnce(git::RemoteCallbacks<'_>
     callbacks.progress = callback
         .as_mut()
         .map(|x| x as &mut dyn FnMut(&git::Progress));
-    let mut get_ssh_key = get_ssh_key; // Coerce to unit fn type
-    callbacks.get_ssh_key = Some(&mut get_ssh_key);
+    let mut get_ssh_key = get_ssh_keys; // Coerce to unit fn type
```
[the remark this answers is not in the document]
Comment (martinvonz) [reply cli/src/commands/git.rs]: Done.

File: lib/src/git.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/git.rs:907]: Thanks, that's much better.

File: lib/src/git.rs [hunk 4]
```diff
@@ -894,24 +895,30 @@ impl<'a> RemoteCallbacks<'a> {
                     // Try to get the SSH key from the agent once. We don't even check if
                     // $SSH_AUTH_SOCK is set because Windows uses another mechanism.
                     if !tried_ssh_agent {
-                        tracing::info!(username, "using ssh_key_from_agent");
```
Comment (martinvonz) [inline lib/src/git.rs:897]: I also changed the message to say "trying ssh_key <key>" and "trying ssh_key_from_agent"

**Pull request merged**